app/models/shipit/hook.rb

```diff
@@ -75,6 +75,13 @@ def emit(event, stack, payload)
           stack_id: stack&.id,
           payload: coerce_payload(payload),
         )
+        deliver_internal_hooks(event, stack, payload)
+      end
+
+      def deliver_internal_hooks(event, stack, payload)
+        (Shipit.internal_hook_receivers || []).each do |receiver|
+          receiver.deliver(event, stack, payload)
```

Comment:
What sort of error handling do we need here? Catch everything, log it, and carry on?

Reply:
You can see it two ways. Either you consider theses internal hooks are part of the app and then if they are borken, it's like your app was broken and you'll fix it soon enough.

Or you can consider them as some kind of third party, and prevent them from breaking the app.

If you do the later, then yeah you'd want to rescue `StandardError`, and log it (and provide some kind of hook point to plug custom exception reporters).

Honestly I think solution 1 is ok, but I wouldn't mind 2.

Reply:
I wondered about this too. I decided not to add the error handling because logging is so often overlooked. Main app code can quite easily rescue exceptions from their `deliver` methods, but that's a decision they can make at the time. If it were likely that multiple teams would be adding a bunch of sophisticated hooks, then more error handling is justified, but I don't see the need for that right now.